nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocument.java

```diff
@@ -16,65 +16,109 @@
  */
 package org.nuxeo.compound.documents;
 
-import static org.junit.Assert.assertFalse;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static javax.servlet.http.HttpServletResponse.SC_SEE_OTHER;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
 import java.io.IOException;
+import java.util.Set;
 
 import javax.inject.Inject;
+import javax.ws.rs.core.MultivaluedMap;
 
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.nuxeo.ecm.core.api.CoreSession;
-import org.nuxeo.ecm.core.api.DocumentModel;
-import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriterTest;
-import org.nuxeo.ecm.core.io.marshallers.json.JsonAssert;
-import org.nuxeo.ecm.core.io.marshallers.json.document.DocumentModelJsonWriter;
-import org.nuxeo.ecm.core.io.registry.context.RenderingContext;
-import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.io.registry.MarshallingConstants;
 import org.nuxeo.ecm.core.test.annotations.Granularity;
 import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.restapi.test.RestServerFeature;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.jaxrs.test.HttpClientTestRule;
 import org.nuxeo.runtime.test.runner.Deploy;
 import org.nuxeo.runtime.test.runner.Features;
 import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.ServletContainerFeature;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.jersey.core.util.MultivaluedMapImpl;
 
 @RunWith(FeaturesRunner.class)
-@Features(CoreFeature.class)
+@Features(RestServerFeature.class)
 @RepositoryConfig(cleanup = Granularity.METHOD)
 @Deploy("org.nuxeo.ecm.platform.video:OSGI-INF/core-types-contrib.xml")
 @Deploy("org.nuxeo.ecm.platform.audio.core:OSGI-INF/core-types-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.tag:OSGI-INF/faceted-tag-service-core-types.xml")
 @Deploy("org.nuxeo.ecm.platform.picture.core:OSGI-INF/picture-schemas-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.types:OSGI-INF/subtypes-enricher-contrib.xml")
 @Deploy("org.nuxeo.compound.documents")
-public class TestCompoundDocument extends AbstractJsonWriterTest.Local<DocumentModelJsonWriter, DocumentModel> {
+public class TestCompoundDocument {
 
     @Inject
-    protected CoreSession session;
-    
-    public TestCompoundDocument() {
-        super(DocumentModelJsonWriter.class, DocumentModel.class);
+    protected ServletContainerFeature servletContainerFeature;
+
+    protected HttpClientTestRule httpClientRule;
+
+    protected ObjectMapper mapper = new ObjectMapper();
+
+    protected Set<String> subtypes = Set.of("File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
```

Comment:
```suggestion
    protected static final Set<String> ALLOWED_COMPOUND_SUBTYPES = Set.of("File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
```

Reply:
After dropping `createSubTypes`, the set is only used here. So this member has been made into a local variable of `checkSubTypes`